src/pages/virtual-machines/modal/RestoreModal.tsx

```diff
@@ -80,6 +80,12 @@ export const RestoreModal = ({ virtualEnvironment }) => {
     dispatch(setNetworkAction([]));
   }, []);
 
+  useEffect(() => {
+    if (clusters.length > 0 && task.restoreClusterId === '') {
```

Comment:
What if restoreClusterId is null or undefined? Check if value of restoreClusterId is falsy value ( !task.restoreClusterId) 

https://developer.mozilla.org/en-US/docs/Glossary/Falsy